src/sources/hawker/index.ts

```diff
@@ -0,0 +1,102 @@
+import { Hawker, HawkerRaw } from './model';
+
+import autoParse from '../../util/autoParse-datagov';
+
+const splitDate = (dateString: string) => {
+  const data = dateString.split('/');
+  if (data.length !== 3) {
+    return undefined;
+  }
+  return {
+    year: data[2],
+    month: data[1].length === 1 ? '0' + data[1] : data[1],
+    day: data[0].length === 1 ? '0' + data[0] : data[0],
+  };
+};
+
+/**
+ * Get the upcoming close date and the reason behind the closure
+ *
+ * assumption is made that the dates are sequential and will not overlap,
+ * hence only enddate is only use to find the closest event
+ */
+const getCloseDetails = (hawker: HawkerRaw) => {
+  const today = new Date();
+  const keys = ['q1', 'q2', 'q3', 'q4', 'others'];
+  const upcomingClosures: { key: string; endDate: Date }[] = [];
+
+  // Convert a date string into a Date object, factoring in SG timezone
+  const convertFromDateString = (dateString: string): Date => {
+    const split = splitDate(dateString);
+    if (!split) {
+      return new Date(new Date(dateString).toLocaleDateString('en-SG'));
+    }
+    return new Date(
+      `${split.year}-${split.month}-${split.day}T00:00:00.000+0800`
+    );
+  };
+
+  for (const key of keys) {
+    const tempKey =
+      key === 'others' ? 'other_works_enddate' : key + '_cleaningenddate';
```

Comment:
nit: use template strings

```suggestion
      key === 'others' ? 'other_works_enddate' : `${key}_cleaningenddate`;
```